src/app/backend/resource/deployment/events.go

```diff
@@ -19,30 +19,32 @@ import (
 	"github.com/kubernetes/dashboard/src/app/backend/resource/dataselect"
 	"github.com/kubernetes/dashboard/src/app/backend/resource/event"
 	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
 	client "k8s.io/client-go/kubernetes"
 )
 
 // GetDeploymentEvents returns model events for a deployment with the given name in the given
 // namespace
 func GetDeploymentEvents(client client.Interface, dsQuery *dataselect.DataSelectQuery, namespace string, deploymentName string) (
 	*common.EventList, error) {
-
-	deployment, err := client.ExtensionsV1beta1().Deployments(namespace).Get(deploymentName, metaV1.GetOptions{})
-	if err != nil {
-		return nil, err
-	}
-
-	selector, err := metaV1.LabelSelectorAsSelector(deployment.Spec.Selector)
+	
+	fieldSelector, err := fields.ParseSelector("involvedObject.name=" + deploymentName)
```

Comment:
Could you use `EventInvolvedNameField` from `k8s.io/kubernetes/pkg/api/field_constants.go`? It will be better than hardcoding this selector.